File: makeArrayConsecutive2.js
function makeArrayConsecutive2(statues) {
    maxNum = minNum = statues[0];
    for (i = 1; i < statues.length; i++){
        if (statues[i] > maxNum){
            maxNum = statues[i]
        } 
        if(statues[i] < minNum){
            minNum = statues[i]
        }
    }
    //this works because I got the min and max num of the array
    //so all of the other nums in the array are inbetween them
    //so if subtract the max from the min and subtract the numbers
    //already inbetween the max and the mind num. Also we dont have
    //to worry about duplicates because of the instructions.
    return (maxNum - minNum - (statues.length - 1))
}

File: makeArrayConsecutinve2/makeArrayConsecutive2.py
def makeArrayConsecutive2(statues):
    max_num = min_num = statues[0]
    for i in range(len(statues)):
        if max_num < statues[i]:
            max_num = statues[i]
        if min_num > statues[i]:
            min_num = statues[i]
            
    return max_num - min_num - (len(statues) - 1)